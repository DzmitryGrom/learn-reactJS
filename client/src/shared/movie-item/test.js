import React from 'react';
import { mount } from 'enzyme';
import Item from './index';

const mockDates =  {
    "id": 0,
    "title": "Dilwale Dulhania Le Jayenge",
    "tagline": "string",
    "vote_average": 0,
    "vote_count": 0,
    "release_date": "1995-02-01",
    "poster_path": "https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
    "overview": "string",
    "genres": ["Comedy", "Drama", "Romance"]
}

describe('tests items', () => {
  it('render with dates', () => {
      const component = mount(<Item  {...mockDates} />);
      expect(component).toMatchSnapshot();
      expect(component.find('.movieItem__title').text()).toBe('Dilwale Dulhania Le Jayenge');
      expect(component.find('.movieItem__genre').text()).toBe("Comedy Drama Romance ");
      expect(component.find('.movieItem__year').text()).toBe('1995');
      expect(component.find('.movieItem__img').prop('style')).toEqual({"backgroundImage": "url(https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg)"});
  });

})