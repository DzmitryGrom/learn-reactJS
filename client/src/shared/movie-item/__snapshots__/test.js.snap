// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests items render empty 1`] = `
<MovieItem
  genres={null}
  poster_path={null}
  release_date={null}
  title={null}
>
  <div
    className="movieItem"
  >
    <div
      className="movieItem__img"
      style={
        Object {
          "backgroundImage": "url(null)",
        }
      }
    />
    <div
      className="movieItem__info"
    >
      <div>
        <h3
          className="movieItem__title"
        />
        <h4
          className="movieItem__genre"
        />
      </div>
      <span
        className="movieItem__year"
      />
    </div>
  </div>
</MovieItem>
`;

exports[`tests items render with dates 1`] = `
<MovieItem
  genres={
    Array [
      "Comedy",
      "Drama",
      "Romance",
    ]
  }
  id={0}
  overview="string"
  poster_path="https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg"
  release_date="1995-02-01"
  tagline="string"
  title="Dilwale Dulhania Le Jayenge"
  vote_average={0}
  vote_count={0}
>
  <div
    className="movieItem"
  >
    <div
      className="movieItem__img"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg)",
        }
      }
    />
    <div
      className="movieItem__info"
    >
      <div>
        <h3
          className="movieItem__title"
        >
          Dilwale Dulhania Le Jayenge
        </h3>
        <h4
          className="movieItem__genre"
        >
          <span
            key="Comedy"
          >
            Comedy
             
          </span>
          <span
            key="Drama"
          >
            Drama
             
          </span>
          <span
            key="Romance"
          >
            Romance
             
          </span>
        </h4>
      </div>
      <span
        className="movieItem__year"
      >
        1995
      </span>
    </div>
  </div>
</MovieItem>
`;
